macOS/debugger.cpp

```diff
@@ -68,12 +68,37 @@ vm_prot_t Debugger::MacOSProtectionFlags(MemoryProtection memory_protection) {
   }
 }
 
+void Debugger::FreeShare(void *address, size_t size) {
+  if (size == 0) {
+    WARN("FreeShare is called with size == 0\n");
+    return;
+  }
+
+  mach_port_t shm_port = share_mem_map[(mach_vm_address_t)address];
+  kern_return_t krt = mach_port_destroy(mach_target->Task(), shm_port);
+  if (krt != KERN_SUCCESS) {
+    FATAL("Error (%s) destroy port for shared memory @ 0x%llx\n", mach_error_string(krt), (mach_vm_address_t)address);
+  }
+
+  krt = mach_vm_deallocate(mach_task_self(), (mach_vm_address_t)address, (mach_vm_size_t)size);
+  if (krt != KERN_SUCCESS) {
+    FATAL("Error (%s) freeing memory @ 0x%llx\n", mach_error_string(krt), (mach_vm_address_t)address);
+  }
+}
+
 void Debugger::RemoteFree(void *address, size_t size) {
+  if (share_mem_map.find((mach_vm_address_t)address) != share_mem_map.end()) {
```

Comment:
Don't have a good place to put this comment, so noting here: In Debugger::OnProcessExit all remaining shared memory objects should be cleared up. Note that (even though we can with shared memory) we shouldn't ever call RemoteRead after OnProcessExit. The earlier comment about LiteCov::OnProcessExit ensures that coverage will be read from the target process before Debugger::OnProcessExit is called.